src/components/AddRestaurant.ts

```diff
@@ -0,0 +1,115 @@
+import RestaurantType from "../type/Restaurant";
+import BottomSheet from "./BottomSheet";
+
+class AddRestaurant extends HTMLElement {
+  private controller;
+
+  constructor() {
+    super();
+    this.controller = globalThis.controller;
+    this.render();
+    this.onClickCancelButton();
+    this.onSubmitRestaurantForm();
+  }
+
+  render() {
+    this.innerHTML = `
+        <h2 class="modal-title text-title">새로운 음식점</h2>
+        <form id="restaurantForm">
+
+          <!-- 카테고리 -->
+          <div class="form-item form-item--required">
+            <label for="category text-caption">카테고리</label>
+            <select name="category" id="category" required>
+              <option value="">선택해 주세요</option>
+              <option value="한식">한식</option>
+              <option value="중식">중식</option>
+              <option value="일식">일식</option>
+              <option value="양식">양식</option>
+              <option value="아시안">아시안</option>
+              <option value="기타">기타</option>
+            </select>
+          </div>
+
+          <!-- 음식점 이름 -->
+          <div class="form-item form-item--required">
+            <label for="name text-caption">이름</label>
+            <input type="text" name="name" id="name" required>
+          </div>
+
+          <!-- 거리 -->
+          <div class="form-item form-item--required">
+            <label for="distance text-caption">거리(도보 이동 시간) </label>
+            <select name="distance" id="distance" required>
+              <option value="">선택해 주세요</option>
+              <option value="5">5분 내</option>
+              <option value="10">10분 내</option>
+              <option value="15">15분 내</option>
+              <option value="20">20분 내</option>
+              <option value="30">30분 내</option>
+            </select>
+          </div>
+
+          <!-- 설명 -->
+          <div class="form-item">
+            <label for="description text-caption">설명</label>
+            <textarea name="description" id="description" cols="30" rows="5"></textarea>
+            <span class="help-text text-caption">메뉴 등 추가 정보를 입력해 주세요.</span>
+          </div>
+
+          <!-- 링크 -->
+          <div class="form-item">
+            <label for="link text-caption">참고 링크</label>
+            <input type="text" name="link" id="link">
+            <span class="help-text text-caption">매장 정보를 확인할 수 있는 링크를 입력해 주세요.</span>
+          </div>
+
+          <!-- 취소/추가 버튼 -->
+          <div class="button-container">
+            <button id="cancelButton" type="button" class="button button--secondary text-caption">취소하기</button>
+            <button type="submit" class="button button--primary text-caption">추가하기</button>
+          </div>
+        </form>
+      `;
+  }
+
+  onClickCancelButton() {
+    const cancelButton = document.getElementById("cancelButton");
```

Comment:
document 수준에서 query를 하면 이 컴포넌트가 하나만 더 렌더되어 있어도 정확하게 엘리먼트를 가져오기 어렵지 않을까요?
```suggestion
    const cancelButton = this.getElementById("cancelButton");
```

Reply:
```
const cancelButton = document.querySelector(
      "#restaurantForm > div > #cancelButton"
    );
```
현재 위와 같이, cancelButton이 restaurantForm안의 취소 버튼만 가리키도록 수정하였습니다. 

this로 스코프를 지정하여 하는 방법은 시도해보았지만 타입 관련 에러가 뜨더라구요.. 
suggested change처럼, typescript에서 this로 element를 가리키는 방법에 대해서 좀 더 자세히 말씀해주실 수 있으실까요?